### DISCUSSION TITLE
0.7 get inherited attributes?

### DISCUSSION THREAD

**Discussion by Upbolt** [leptos-rs/leptos#2922] [Q&A] [answered]
in 0.6 you were able to do 
```rs
#[component]
fn Foo(
  #[prop(attrs)] attributes: Vec<(&'static str, Attribute)>
) -> impl IntoView
```
but this doesn't appear to be the case with 0.7 using `Vec<AnyAttribute>`, i get back an empty vec

**Comment by gbj** [comment, marked as answer]
In 0.7 you can just add attributes to a component and they will be added to the element(s) it returns. See the `spread` example. 

Feel free to provide more details if this does not suit your needs.

**Reply by Upbolt** [reply]
does this also apply when you use the builder pattern instead of the view macro?

**Reply by Upbolt** [reply]
nevermind, just tested it and it seems like it does 🎉

```rs
use leptos::{html::input, prelude::*};

fn main() {
    mount_to_body(|| {
        view! {
            <App />
        }
    });
}

#[component]
fn App() -> impl IntoView {
    view! {
        <Foo {..} aria-valuenow="hello" />
    }
}

#[component]
fn Foo() -> impl IntoView {
    input::<Dom>()
}
```

**Reply by gbj** [reply]
Yeah, the advantage of the new renderer is that this is done at the type level, not with a macro, so it works either way.

If you want builder all the way this should work too (didn't check it, sorry if it doesn't compile)
```rust
#[component]
fn App() -> impl IntoView {
    Foo().attr("aria-valuenow", "hello")
}

#[component]
fn Foo() -> impl IntoView {
    input::<Dom>()
}
```
